mocks/sender.go

```diff
@@ -8,29 +15,29 @@ import (
 )
 
 type MockSender struct {
+	mu                  *sync.Mutex
 	totalSent           int
 	totalCalls          int
 	failureIndexesOrder [][]int64
 	firstCallEvents     []*pb.Event
 	eventsIdsFreqs      map[string]int
-	mutex               *sync.Mutex
 }
 
 func NewMockSender() *MockSender {
 	return &MockSender{
 		failureIndexesOrder: [][]int64{},
 		eventsIdsFreqs:      map[string]int{},
-		mutex:               &sync.Mutex{},
+		mu:                  &sync.Mutex{},
 	}
 }
 
 func (ms *MockSender) SendEvents(
 	ctx context.Context,
 	events []*pb.Event,
 ) []int64 {
-	ms.mutex.Lock()
-	defer ms.mutex.Unlock()
+	ms.mu.Lock()
 	defer func() { ms.totalCalls++ }()
+	defer ms.mu.Unlock()
```

Comment:
Suggestion: add the Unlock call to the func above?

Reply:
done